### DISCUSSION TITLE
MQTT does not work from Pow K+ to home assistant, MQTT Explorer can connect to Home Assistant

### DISCUSSION THREAD

**Discussion by JanSjo** [UtilitechAS/amsreader-firmware#689] [Q&A] [answered]
I cannot get the connection MQTT to homeassistant to start working.

Probably som simple error in setup.

my first MQTT.

I tried connecting via MQTT Explorer to Home Assistant and that works and I can see in log.

But Pow K+ only show red and "MQTT: Connection failed"

is ther any picutre with example on how everything should be filled in?

I an thinking there should or not should be HTTP:/  berfor IP address etc - so I am missing it ther.

ideas?

**Comment by ArnieO** [comment]
Please post your current configuration, it will make it easier to pinpoint the problem.

**Reply by JanSjo** [reply]
here are some pictures of my configuration first picture my POWk+:

![image](https://github.com/UtilitechAS/amsreader-firmware/assets/6060166/e94d1f50-8583-4109-bdf7-2c15b8a177da)

in home assistant 
![image](https://github.com/UtilitechAS/amsreader-firmware/assets/6060166/6b3eeaf3-e9c9-4f35-946c-d622db6f0292)

![image](https://github.com/UtilitechAS/amsreader-firmware/assets/6060166/65446a90-bee3-4db2-8e08-0a0a2ecaea00)

![image](https://github.com/UtilitechAS/amsreader-firmware/assets/6060166/756a7ba2-1d19-4f39-af71-ddfcd9cc004c)

MQTT Explorer which can connect:
![image](https://github.com/UtilitechAS/amsreader-firmware/assets/6060166/8edd584b-63a0-4325-9883-7540fd1972fd)

**Reply by ArnieO** [reply]
Those IP addresses 192.168.1.xxx : That must be your anonymization of the actual IP addresses, right?
I think you need to show what you have there, as the issue might be there.
They are anyway just internal IPs on your local Wi-Fi, so there should be no harm or risk in disclosing them.

**Reply by JanSjo** [reply]
yes  this is how it looks like

left .112 is HA and right is PowK+ .210

and in MQTT explorer I connect to .112 and it looks like second picture.

probably I am dong something simple error - being first time with MQTT

![image](https://github.com/UtilitechAS/amsreader-firmware/assets/6060166/c7289702-3275-46f7-afd5-ed8197883c59)

below after Connection via MQTT
before login to MQTT in HA
![image](https://github.com/UtilitechAS/amsreader-firmware/assets/6060166/fe48a3b3-e8fa-456c-95da-8149fbdc01db)

After connection:

![image](https://github.com/UtilitechAS/amsreader-firmware/assets/6060166/dd3d3958-d1a8-4cc9-9d17-ac54f84f8c2a)

**Reply by ArnieO** [reply]
I hope someone else in the community sees this, as I am not using HA myself - so can myself only provide limited guidance here.
It looks like you're running a standalone Modsqitto MQTT broker?
My impression is that the commonly used solution for HA is to use the MQTT broker that is embedded in HA?

So... I asked for assistance: https://github.com/UtilitechAS/amsreader-firmware/issues/691

**Reply by simenovrebo** [reply, marked as answer]
![IMG_0622](https://github.com/UtilitechAS/amsreader-firmware/assets/4172901/be0615e6-0b6d-4ca5-87ea-bf6f922db0a1)
Hi,

Try removing the http:/// and just leave the IP in the PowK settings. :-) My settings as example in screen dump.

**Reply by ArnieO** [reply]
Ah... yes - thank you! I should have spotted that. The IP addresses should be without "http://"

**Reply by simenovrebo** [reply]
It's a pleasure. Sometimes help with an extra set of eyes to look at a problem. 😊

Would also leave the field "Hostname for URL" blank to use the mDNS record for the POW-K+ or put the POW-K+ IP address in without the http://. Can also be www.google.com if you want probably. 😉 "Hostname for URL" is used by the "BESØK" link at least as shown in the screen dump below which in my case sends me to the POW-U+ web UI:

<img width="353" alt="Skjermbilde 2023-12-09 kl  08 12 37" src="https://github.com/UtilitechAS/amsreader-firmware/assets/4172901/734803cd-a972-4dc7-a0b7-541fd9733a93">

**Reply by JanSjo** [reply]
Thanks I fiddled around with your suggestions and tried skipping payload homeassistant, just using JSON and tried listening manually in Home assistant and then it started to work.

I then changed back to home assistant payload and change URL etc as in picture and now I get the data :-) .

For the energy graph in HA I used "current hours used" sensor - is that what you would use?

also I added a name in nametag "PowkWhName" - which now is in all sensor names in HA and if I change it to "PowKwh" it is still  "PowkWhName" in HA, added for test - guess I will change manually in HA.

**thanks a lot for the help**

![image](https://github.com/UtilitechAS/amsreader-firmware/assets/6060166/e40f35fd-11f3-4b3d-b8a5-ae338c645d9e)

energy graph in HA
![image](https://github.com/UtilitechAS/amsreader-firmware/assets/6060166/83b76e59-3756-4046-99cf-7cd7e1c347c5)

draft dashboard to see data
![image](https://github.com/UtilitechAS/amsreader-firmware/assets/6060166/6942eb26-7f89-4dd8-a1dd-c8f0c207dd58)

**Reply by simenovrebo** [reply]
Use Accumulated active import.

this guide is probably still relevant: https://www.amsleser.no/module/ets_blog/blog?id_post=4

**Reply by ArnieO** [reply]
@JanSjo I believe you will notice there is a 1-hour shift in the data. This is due to the Norwegian HAN-payload format from NVE, where accumulated meter data (kWh) is only sent once per hour, around 10 seconds after whole hour. The consequence is that HA registers the data in the wrong hour-slot. We do not see any way to prevent that from our side.

**Reply by JanSjo** [reply]
ok, thanks for that info regarding delay.

I checked using current hour vs accumulated change / meter value and change.
it is indeed delay with one hour, but the graph using sensor current hour is just 5 min wrong 1 of 12 values, so it seems ok for overview and, using the accumulated can be used for exact measurement although one hour delayed

![image](https://github.com/UtilitechAS/amsreader-firmware/assets/6060166/27de3482-aa77-432c-a2d5-a5dd430d4b2f)

**Reply by ArnieO** [reply]
Yeah, if you search old "Issues" here you will find several mentions of that 1 hour delay - so I just thought you should be notified.
